### ISSUE TITLE
Possible copyright infringements

### ISSUE THREAD

**Issue by rbalint** [xbmc/chorus2#179]
Please state the license of Chorus2. Since it is based on Kodi it needs to be GPL2+.

Please state the license and origin for each file included in the tarball/repository.
There are many files with protential license problems:
dist/themes/base/images/google-ico.png and dist/themes/base/images/imdb-ico.png may not be legally usable in GPL sofware and I can't tell the source.

dist/images/fanart_default/amp.jpg has the following history:
```
commit 67ea92f9d742e37100760f707e077c4a7eaf4f18
Author: Jeremy <mail@jez500.com>
Date:   Mon Nov 7 21:40:29 2016 +1100

    Updated default backgrounds with CC images

 dist/images/fanart_default/amp.jpg       | Bin 0 -> 185756 bytes
```
Some Creative Commons licenses are not compatible with GPL2+ and I can't tell the license and the exact source of those images.

Including those files in Chorus2 and Kodi makes Chrous2 illegal to distribute if their copyright is not compatible with GPL2+.

Please replace the files with incompatible license with ones legally redistributable with GPL2+ and state their license and origin.

**Comment by MartijnKaijser** [comment]
@jez500 could you look at this?

**Comment by jez500** [comment]
Hey @rbalint & @MartijnKaijser 

I have added this https://github.com/xbmc/chorus2/blob/master/src/lang/en/license.md and swapped out the google/imdb logos for fontawesome icons, hopefully that should cover this?

Let me know if anything else is required

**Comment by rbalint** [comment]
Hi @jez500 & @MartijnKaijser,

Thanks for working on resolving the licensing issues. The CC0 license for the images is OK (compatible with Kodi's GPLv2+ and the libs). 

The CC-SA-BY 3.0 for the whole addon is not compatible with GPLv2+ thus it can't be part of Kodi.
https://www.gnu.org/licenses/license-list.html#ccby
CC licenses are generally not for programs.
I suggest relicensing the addon under GPLv2+ to keep licensing simple and legally valid.

I'm not fully convinced that the fontawesome icon's SIL OFL allows bundling of the icons to GPLv2+ programs this way, especially since Kodi's GPL does not have the fonts exception, but someone from the Kodi Team who works on licensing issues may also comment on that.
http://fontawesome.io/license/

**Comment by jez500** [comment]
Hi @rbalint 

TBH I know nothing about licensing and just copied what was used for a core skin: https://github.com/xbmc/xbmc/blob/master/addons/skin.estouchy/LICENSE.txt 

If you can tell me what to add, or better, provide a PR :) then I will update accordingly

**Comment by natethomas** [comment]
@rbalint I'm not sure if compatibility matters in this situation. Chorus2 is not being compiled inside Kodi, so it's not dynamically linked. It's its own application communicating with Kodi through JSON-RPC. As such, Chorus2 and Kodi do not need to have compatible licenses, as far as I understand it.

**Comment by rbalint** [comment]
@natethomas Dynamic linking is something to consider when you try to mix GPL and non-GPL programs which are not distributed together.
GPL allows 'bundling' your program with other programs with incompatible licenses, but Chorus2 is distributed as _part_ of Kodi which makes GPL compatible licensing necessary.
The proof of Chorus2 being tightly coupled significant part of Kodi is that Kodi does not even start if I remove addons/webinterface.default/* where Chorus2 resides.
Disclaimer: I'm not a lawyer and this is not a legal advice.

**Comment by rbalint** [comment]
@jez500 OK, I'll prepare a PR with the relicensing of the code under GPL2+
I suggest replacing the icons with text links in the meantime, unless GPL2+ icons are found.

**Comment by natethomas** [comment]
@rbalint, I believe you are looking at that backwards. Kodi also doesn't start in Windows if you remove Windows or in Linux if you remove Linux. Whether something is GPL is determined by whether it is derivative of the source, not whether the source is dependent upon it. The fact that Kodi depends on the separate Chorus2 application should not actually affect the Chorus2 application.

**Comment by rbalint** [comment]
@natethomas Chorus2 is now embedded in Kodi's source unlike Linux or Windows. I suggest seeking legal advice for Kodi at https://www.softwarefreedom.org/services/ if you want mix the code with parts distributed under licenses which are not well known to be compatible with GPLv2+.
SFLC  provides pro bono services for Free Software projects.

**Comment by natethomas** [comment]
@rbalint, yup, they are actually our lawyers. We have not asked them about this particular combination of software, but I did attend their recent fall conference, in which compliance was discussed fairly heavily. There, the idea was stressed fairly heavily that, because there are no actual court verdicts on the exact interpretation of the GPL, and thus any of the three major interpretations of derivative works may ultimately be the correct one, as long as the major copyright holders agree, any reasonable interpretation will probably work. 

In other words, if the XBMC Foundation, @jez500, and whoever was the primary author of the API being used by Chorus2 within Kodi don't object to Chorus2 being CC under some kind of reasonable interpretation of derivative works of the GPL, then there simply isn't anything worth discussing. Perhaps a future time may come when a court makes a final verdict interpreting exactly where the boundaries of derivative works lie, or an author of this portion of the codebase may strongly object, believing chorus2 to be derivative of previous GPL software, but until that point, this should be simply a non-issue.

With that said, if you ARE an author of Chorus2 or the API being used by Chorus2, that's a different thing.

**Comment by rbalint** [comment]
@natethomas I raised the issue regarding Chorus2 being CC 3.0 licensed _not_ because it can be considered a derivative of Kodi, but because it is _integrated_ into Kodi. Writing CC 3.0 addons is fine from legal POV IMO if they are not distributed within Kodi.
Since @jez500 is OK with relicensing under GPL2+ this issue will be solved unquestionably, but license compatibility is something to consider when integrating new parts to Kodi.

**Comment by jez500** [comment]
Solved with #208

**Comment by rbalint** [comment]
@jez500 Please reopen the issue, there are still some files which are problematic. But we are close. :-)

Are you the author of src/lib/required/cocktailMixins/sorted.js ? There is no copyright but it is src/lib which suggests it having 3rd party source.

dists/screenshots/* contains images which are clearly copyrighted and claiming fair use is always fishy. You can create screenshots with material in public domain http://thecreatorsproject.vice.com/blog/10000-film-clips-now-available-for-free

**Comment by jez500** [comment]
Hey @rbalint - reopened 

sorted is this https://github.com/gitterHQ/sorted but I didn't actually end up using it so will just remove it from the codebase

screenshots - i'll recreate when I find some time to recreate a media db

**Comment by MartijnKaijser** [comment]
Is still relevant?

**Comment by rbalint** [comment]
@MartijnKaijser @jez500 I believe so, the screenshots still include copyrighted material.
